packages/shared/lib/src/sdk/mod.rs

```diff
@@ -296,42 +223,54 @@ impl Sdk {
                 let info = tx_response.info.to_string();
                 let log = tx_response.log.to_string();
 
-                let response = TxResponse {
-                    code,
-                    commitments: batch_tx_results,
-                    gas_used,
-                    hash,
-                    height,
-                    info,
-                    log,
-                };
+                let response =
+                    tx::TxResponse::new(code, batch_tx_results, gas_used, hash, height, info, log);
                 to_js_result(borsh::to_vec(&response)?)
             }
             _ => return Err(JsError::new(&format!("Tx not applied: {}", &hash))),
         }
     }
 
     /// Build a batch Tx from built transactions and return the bytes
-    pub fn build_batch(built_txs: Vec<BuiltTx>) -> Result<BuiltTx, JsError> {
+    pub fn build_batch(txs: JsValue) -> Result<JsValue, JsError> {
+        let mut built_txs: Vec<tx::Tx> = vec![];
+        let built_txs_bytes: Vec<Vec<u8>> = txs.into_serde().unwrap();
+
+        for bytes in built_txs_bytes.iter() {
+            let tx: tx::Tx = borsh::from_slice(&bytes)?;
+            built_txs.push(tx);
+        }
+
+        // Get wrapper args
+        let first_tx = built_txs
+            .iter()
+            .nth(0)
```

Comment:
You can also just `.get(0)` without converting vec to iter :)

Reply:
Thanks! I updated it in a couple places. Just re-tested `Withdraw`, working fine!